pinot-core/src/main/java/org/apache/pinot/core/query/reduce/ExecutionStatsAggregator.java

```diff
@@ -117,6 +118,13 @@ public synchronized void aggregate(@Nullable ServerRoutingInstance routingInstan
     String operatorId = metadata.get(DataTable.MetadataKey.OPERATOR_ID.getName());
     if (operatorId != null) {
       _operatorIds.add(operatorId);
+      if (_enableTrace) {
+        Map<String, String> metadataWithoutOperatorId = new HashMap<>(metadata);
+        metadataWithoutOperatorId.remove(DataTable.MetadataKey.OPERATOR_ID);
```

Comment:
no need to remove the key itself IMO. it is not that much of an overhead compared to doing a map copy.

Reply:
Only reason is because the response looks unclean. The key is already operatorID and inside the map we put operatorId again.